extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -378,32 +408,76 @@ export class Bind {
       });
 
       const onPop = () => this.setState(oldState);
-      return this.setState(result)
-        .then(() => this.getDataForHistory_())
-        .then(data => {
-          this.history_.push(onPop, data);
+      return this.setState(result).then(() => {
+        return this.getDataForHistory_().then(data => {
+          this.pushToHistoryQueue_({op: HistoryOp.PUSH, data, onPop});
         });
+      });
+    });
+  }
+
+  /**
+   * @param {!HistoryTaskDef} task
+   * @private
+   */
+  pushToHistoryQueue_(task) {
+    let enqueue = true;
+
+    // Collapse if consecutive "replace" operations.
+    if (task.op === HistoryOp.REPLACE) {
+      const lastTask = this.historyQueue_[this.historyQueue_.length - 1];
+      if (lastTask && lastTask.op === HistoryOp.REPLACE) {
```

Comment:
Nit: Check that there's at least 1 queued item before, instead of doing a missed lookup.